Plutarch/Internal/Term.hs

```diff
@@ -320,24 +322,28 @@ we want to trace, and if so, under what log level and mode.
 
 @since 1.6.0
 -}
-newtype Config = Config (Last (LogLevel, TracingMode))
-  deriving
-    ( -- | @since 1.6.0
-      Semigroup
-    , -- | @since 1.6.0
-      Monoid
-    )
-    via (Last (LogLevel, TracingMode))
+data Config = Config
+  { trace'config :: Last (LogLevel, TracingMode)
+  , constHoist'config :: Last Integer
+  }
   deriving stock
     ( -- | @since 1.6.0
       Eq
     , -- | @since 1.6.0
       Show
     )
 
+-- | @since WIP
+instance Semigroup Config where
+  (Config a b) <> (Config a' b') = Config (a <> a') (b <> b')
+
+-- | @since WIP
+instance Monoid Config where
+  mempty = Config mempty (Last $ Just 20)
```

Comment:
I'm not sure about `Last $ Just 20` here. Now we have `NoTracing <> mempty /= NoTracing`. It'd need to be a bit more elaborate, like `InliningConfig = AlwaysInline | InlineIfSmaller Integer` and when you actually want to extract the config from Last then you can do the default with `fromMaybe (InlineIfSmaller 20) constHoist'config`. Then you can just put `mempty` here to get lawful monoid back.

TLDR; `Nothing` cause `Last` was not set must be different than "no inline threshold"

Reply:
Ah, yeah. I originally had `mempty` as `Last Nothing`, which would've worked. I'll fixed this

Reply:
I was not feeling like rolling up a whole new type for just this. I just made `consthoist'config :: Last Integer` and made `-1` represent "never hoist"

Reply:
Magic values :disappointed:

Reply:
ehh I think this is pretty sensible